### DISCUSSION TITLE
Network Visualization

### DISCUSSION THREAD

**Discussion by jenapkst** [openthread/openthread#9545] [Q&A] [answered]
Hi,

I want to arrange the nodes of the network at any node point. 
Retrieved "router table" data from a node (Leader). 
"Next Hop" data is not clear understanding. Mostly I find two nodes are indicating each other as next hop.

Any doc, any tools or any information for me to workout?

regards
PK Jena

**Comment by abtink** [comment, marked as answer]
Some suggestions/recommendations:
- Can check [the `meshdiag` APIs here](https://github.com/openthread/openthread/blob/6bfe59dabf1cf4c2b02d950418a218a6a030d6c2/include/openthread/mesh_diag.h)
- And the related [CLI commands](https://github.com/openthread/openthread/blob/6bfe59dabf1cf4c2b02d950418a218a6a030d6c2/src/cli/README.md#meshdiag-topology-ip6-addrschildren)
- They can be used to discover topology and information about all nodes within mesh.
- Also may help to check `otThreadGetNextHopAndPathCost()` API and its related `nexthop` CLI command.

**Reply by jenapkst** [reply]
Thank you Abtink for information.

regards
PK Jena
